docs/en/04_Changelogs/4.0.0.md

```diff
@@ -108,6 +109,65 @@ Core template locations have moved - if you're including or overriding these
 no longer exists, and instead template locations will be placed in paths that match
 the `SilverStripe\Forms` namespace.
 
+#### Upgrade module paths in file references
+
+You should no longer rely on modules being placed in a deterministic folder (e.g. `/framework`),
+and use getters on the [Module](api:SilverStripe\Core\Manifest\Module) object instead.
+They expect a module composer name, followed by the module relative path, separated by a double colon. 
+This prepares SilverStripe for moving modules out of the webroot at a later point.
+
+Usage in templates:
+
+```ss
+<!-- before -->
+<img src="framework/images/image.png" />
+<% require css("framework/css/styles.css") %>
+
+<!-- after -->
+<img src="$ModulePath(silverstripe/framework)/image.png" />
+<% require css("silverstripe/framework: css/styles.css") %>
+```
+
+Usage in Requirements:
+
+```php
+// before
+Requirements::css('framework/css/styles.css');
+
+// after
+Requirements::css('silverstripe/framework: css/styles.css');
+```
+
+Usage with custom logic:
+
+```php
```

Comment:
Parsedown doesn't do syntax highlighting? We're using https://github.com/google/code-prettify as a JS lib, and that doesn't do diff: https://github.com/google/code-prettify/issues/62. Ideas?